core/src/main/scala/kafka/log/Log.scala

```diff
@@ -122,22 +124,29 @@ class Log(@volatile var dir: File,
 
   /* the actual segments of the log */
   private val segments: ConcurrentNavigableMap[java.lang.Long, LogSegment] = new ConcurrentSkipListMap[java.lang.Long, LogSegment]
+
+  val topicPartition: TopicPartition = Log.parseTopicPartitionName(dir)
+
+  val leaderEpochCache: LeaderEpochCache = new LeaderEpochFileCache(topicPartition, () => logEndOffsetMetadata,
+    new LeaderEpochCheckpointFile(LeaderEpochFile.newFile(dir)))
```

Comment:
At this point, the log dir may not have been created. So, we probably need to make sure the log dir exists first.